web/src/components/pages/contribution/sentence-collector/write/sentence-input-and-rules/sentence-input-and-rules.tsx

```diff
@@ -46,15 +47,21 @@ export const SentenceInputAndRules: React.FC<Props> = ({
   citation,
   error,
   variantTokens,
+  variantNames,
   selectedVariant,
   mode,
 }) => {
+  const { l10n } = useLocalization()
   const isSentenceError =
     error && error?.type !== SentenceSubmissionError.NO_CITATION
   const isCitationError = error?.type === SentenceSubmissionError.NO_CITATION
   const hasVariants = variantTokens && variantTokens.length > 0
-
-  const { l10n } = useLocalization()
+  const variantOptions: string[] = [l10n.getString('sentence-variant-select-multiple-variants')]
+  if (hasVariants) {
+    for (let i=0; i < variantTokens.length; i++) {
+      variantOptions.push(`${variantNames[i]} [${variantTokens[i]}]`)
```

Comment:
Could you use Array.map here instead? You can then spread the result from the map into the variantOptions array. Generally, we prefer to use Array functions that `map` that don't mutate variables unlike `push` which does.

For example,

```js
const resultFromMap = variantTokens.map(...) // example variable name feel free to use one that makes sense to you

const variantOptions = [l10n.getString('sentence-variant-select-multiple-variants'), ...resultFromMap]
```